dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/MinMaxDataElementController.java

```diff
@@ -0,0 +1,225 @@
+        }
+
+        List<MinMaxDataElement> minMaxDataElements = minMaxService.getMinMaxDataElements( query );
+
+        RootNode rootNode = NodeUtils.createMetadata();
+
+        if ( !query.isSkipPaging() )
+        {
+            query.setTotal( minMaxService.count( query ) );
+            rootNode.addChild( NodeUtils.createPager( query.getPager() ) );
+        }
+
+        rootNode.addChild( fieldFilterService.filter( MinMaxDataElement.class, minMaxDataElements, fields ) );
+
+        return rootNode;
+    }
+
+    //--------------------------------------------------------------------------
+    // POST
+    //--------------------------------------------------------------------------
+
+    @RequestMapping( method = RequestMethod.POST, consumes = "application/json" )
+    public void postJsonObject( HttpServletRequest request, HttpServletResponse response )
+        throws Exception
+    {
+        MinMaxDataElement minMax = deserializeJsonEntity( request, response );
+
+        validate( minMax );
+
+        minMax = getReferences( minMax );
+
+        MinMaxDataElement persisted = minMaxService.getMinMaxDataElement( minMax.getSource(), minMax.getDataElement(), minMax.getOptionCombo() );
+
+        if ( Objects.isNull( persisted ) )
+        {
+            minMaxService.addMinMaxDataElement( minMax );
+        }
+        else
+        {
+            persisted.mergeWith( minMax );
+            minMaxService.updateMinMaxDataElement( persisted );
+        }
+
+        WebMessage webMessage = new WebMessage();
+        webMessage.setHttpStatus( HttpStatus.CREATED );
+        webMessage.setStatus( Status.OK );
+
+        webMessageService.send( webMessage, response, request );
+    }
+
+    //--------------------------------------------------------------------------
+    // DELETE
+    //--------------------------------------------------------------------------
+
+    @RequestMapping( method = RequestMethod.DELETE, consumes = "application/json" )
+    public void deleteObject( HttpServletRequest request, HttpServletResponse response ) throws Exception
+    {
+
+        MinMaxDataElement minMax = deserializeJsonEntity( request, response );
+
+        validate( minMax );
+
+        minMax = getReferences( minMax );
+
+        MinMaxDataElement persisted = minMaxService.getMinMaxDataElement( minMax.getSource(), minMax.getDataElement(), minMax.getOptionCombo() );
+
+        if ( Objects.isNull( persisted ) )
+        {
+            throw new WebMessageException( WebMessageUtils.notFound( "Can not find MinMaxDataElement." ) );
+        }
+
+        minMaxService.deleteMinMaxDataElement( persisted );
+
+        webMessageService.send( WebMessageUtils.ok( "MinMaxDataElement deleted." ), response, request );
+    }
+
+    private MinMaxDataElement deserializeJsonEntity( HttpServletRequest request, HttpServletResponse response ) throws IOException
+    {
+        return renderService.fromJson( request.getInputStream(), MinMaxDataElement.class );
```

Comment:
Don't really need a method for this? this is done in AbstractCrudController so that it can be overridden, but i don't think this is useful here..